cmd/server/main.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/redhatinsights/insights-operator-conditional-gathering/pkg/service"
+	"github.com/redhatinsights/insights-operator-conditional-gathering/pkg/service/transport"
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	var (
+		// Flags
+		httpAddr = flag.String("httpAddr", ":8080", "http listen address")
+
+		httpServer *http.Server
+	)
+
+	flag.Parse()
+
+	// Logger
+	var logger log.Logger
+	logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC)
+
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Repository
+	dataPath := os.Getenv("RULES_DATA_PATH")
+	repo := service.NewRepository(dataPath)
+
+	// Services
+	svc := service.New(repo)
+
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt, syscall.SIGTERM)
+	defer signal.Stop(interrupt)
+
+	g, ctx := errgroup.WithContext(ctx)
+
+	// HTTP
+	g.Go(func() error {
+		httplogger := log.With(logger, "component", "http")
+
+		mux := http.NewServeMux()
+		// Access Control and CORS
+		http.Handle("/", accessControl(mux))
+
+		// Handlers
+		mux.Handle("/", transport.NewHTTPHandler(svc, httplogger))
+
+		logger.Log("transport", "http", "address", *httpAddr, "msg", "listening") // nolint: errcheck
+
+		httpServer = &http.Server{
+			Addr:         *httpAddr,
+			ReadTimeout:  10 * time.Second,
+			WriteTimeout: 10 * time.Second,
+		}
+
+		err := httpServer.ListenAndServe()
+		if err != http.ErrServerClosed {
+			return err
+		}
+		return nil
+	})
+
+	// @TODO Implement gRPC
```

Comment:
Why do we need gRPC? I don't think we're passing too much data, so http should be fine

Reply:
Yes let's stick with REST & HTTP for now please.